src/store/accounts/nonce.ts

```diff
@@ -31,10 +32,18 @@ export const currentAccountNonceState = atom(get => {
   const confirmedTransactions = get(currentAccountConfirmedTransactionsState);
   const pendingTransactions = get(currentAccountMempoolTransactionsState);
   const lastLocalNonce = get(currentAccountLocalNonceState);
+  const latestLocallySumbmittedNonce = get(currentAccountLocallySubmittedLatestNonceState);
 
   const apiNonce = get(lastApiNonceState);
-  // We try to use the api nonce first since it will be the most accurate value
-  if (typeof apiNonce === 'number') return apiNonce;
+  if (typeof apiNonce === 'number') {
+    if (latestLocallySumbmittedNonce) {
```

Comment:
typo

Reply:
Note that the api might still return a value but an incorrect value (because the prod api has not picked up the latest tx). So it won't use the local nonce calculation. This won't solve this issue https://github.com/hirosystems/devops/issues/797 but it will solve 'api down' issues and the wallet can correctly build new tx. 👍